lock-impl/src/main/java/com/palantir/lock/impl/LockThreadInfoSnapshotManager.java

```diff
@@ -52,6 +53,9 @@ public LockThreadInfoSnapshotManager(
 
     public void start() {
         scheduleRun();
+        threadInfoConfiguration.subscribe(newThreadInfoConfiguration -> {
```

Comment:
Suppose it's already enabled. What happens if I change the snapshot interval millis, but not whether it's enabled or not? 

Alternatively, what happens if I disable and re-enable data collection within the interval of a threadInfoSnapshotIntervalMillis?

Reply:
Also, subscriptions returns disposables, which you should clean up afterwards on close!